### ISSUE TITLE
Megafusion: error when the fusions inspector report is empty

### ISSUE THREAD

**Issue by ToonRosseel** [nf-core/rnafusion#366]
### Description of the bug

I've got an exit status 1 of the pipeline (v2.3.4) for the megafusion command when the fusions inspector report of a sample contains no fusions. Could this be taken into account when updating the pipeline in the future? All the best
```
Error executing process > 'NFCORE_RNAFUSION:RNAFUSION:FUSIONINSPECTOR_WORKFLOW:MEGAFUSION ($SAMPLE)'

Caused by:
  Process `NFCORE_RNAFUSION:RNAFUSION:FUSIONINSPECTOR_WORKFLOW:MEGAFUSION ($SAMPLE)` terminated with an error exit status (1)

Command executed:

  megafusion.py --fusioninspector $SAMPLE.FusionInspector.fusions.tsv --fusionreport index.html --sample $SAMPLE --out $SAMPLE.vcf
  
  cat <<-END_VERSIONS > versions.yml
  "NFCORE_RNAFUSION:RNAFUSION:FUSIONINSPECTOR_WORKFLOW:MEGAFUSION":
      python: $(python --version | sed 's/Python //g')
  END_VERSIONS

Command exit status:
  1

Command output:
  (empty)

Command error:
  INFO:    Environment variable SINGULARITYENV_TMPDIR is set, but APPTAINERENV_TMPDIR is preferred
  INFO:    Environment variable SINGULARITYENV_NXF_DEBUG is set, but APPTAINERENV_NXF_DEBUG is preferred
  Traceback (most recent call last):
    File "/user/gent/415/vsc41590/.nextflow/assets/nf-core/rnafusion/bin/megafusion.py", line 208, in 
      sys.exit(main())
               ^^^^^^
    File "/user/gent/415/vsc41590/.nextflow/assets/nf-core/rnafusion/bin/megafusion.py", line 204, in main
      megafusion(args.fusioninspector, args.fusionreport, args.sample, args.out)
    File "/user/gent/415/vsc41590/.nextflow/assets/nf-core/rnafusion/bin/megafusion.py", line 195, in megafusion
      write_vcf(column_manipulation(merged_df), header_def(sample), out)
    File "/user/gent/415/vsc41590/.nextflow/assets/nf-core/rnafusion/bin/megafusion.py", line 164, in write_vcf
      df_to_print[
    File "/usr/local/lib/python3.11/site-packages/pandas/core/frame.py", line 3811, in __getitem__
      indexer = self.columns._get_indexer_strict(key, "columns")[1]
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/usr/local/lib/python3.11/site-packages/pandas/core/indexes/base.py", line 6113, in _get_indexer_strict
      self._raise_if_missing(keyarr, indexer, axis_name)
    File "/usr/local/lib/python3.11/site-packages/pandas/core/indexes/base.py", line 6176, in _raise_if_missing
      raise KeyError(f"{not_found} not in index")
  KeyError: "['INFO', 'Sample'] not in index"
```

**Comment by marchoeppner** [comment]
Second this, seeing the same thing on my current analysis run - and no way to get past it without hacking the workflow.

**Comment by Yakun-Pang** [comment]
Run into exactly the same issue as @mschuber there.
At first, the pipeline failed at the fusion report step because of an empty sample results.. After removing the filter that only keep the fusion genes that had been captured by more than two tools, I run into this issue.

**Comment by gurpreet-bioinfo** [comment]
Hi, I also had same issue with v2.3.4 and followed @mschubert 's solution for the first error and now getting megafusion error.

Any luck with solving this error @mschubert @Yakun-Pang @ToonRosseel @marchoeppner ? Thanks.

**Comment by efgstone** [comment]
I am also experiencing the same issue highlighted by @mschuber. I have not attempted the fix for this yet as it seems to only push the error further down the workflow based on the comments posted here.

**Comment by efgstone** [comment]
In case it helps anyone else, I have modified the code in bin/megafusion.py on my instance to fix the ` KeyError: "['INFO', 'Sample'] not in index"` message when no variants were found. I modified the column_manipulation function to add `df["INFO"] = ""` and `df["Sample"] = ""` on lines 125 and 126 respectively. The top of the function on my instance now looks like this:

```
def column_manipulation(df):
    df["ALT"] = ""
    df = df.reset_index()
    df["FORMAT"] = "GT:DV:RV:FFPM"
    df["ID"] = "."
    df["QUAL"] = "."
    df["FILTER"] = "PASS"
    df["REF"] = "N"
    df["INFO"] = ""
    df["Sample"] = ""

    for index, row in df.iterrows():
```

Maybe this could be incorporated into a subsequent release of the pipeline?
